e2e test: compile error if only the tag "ibmcloud" is specified

When attempting to run e2e tests for the ibmcloud provider only, I set tag as ibmcloud only:
```
CLOUD_PROVIDER=ibmcloud GOFLAGS="-tags=ibmcloud" make test-e2e
```

However, I got following compile error:
```
test/e2e/provision.go:80:22: undefined: NewAzureCloudProvisioner
test/e2e/provision.go:82:22: undefined: NewLibvirtProvisioner
note: module requires Go 1.18
FAIL	github.com/confidential-containers/cloud-api-adaptor/test/e2e [build failed]
FAIL
```

--- a/test/e2e/libvirt_test.go
+++ b/test/e2e/libvirt_test.go
@@ -1,4 +1,4 @@
-//go:build libvirt
+//go:build libvirt && cgo
 
 package e2e
 

--- a/test/provisioner/provision_azure.go
+++ b/test/provisioner/provision_azure.go
@@ -1,5 +1,3 @@
-//go:build azure
-
 // (C) Copyright Confidential Containers Contributors
 // SPDX-License-Identifier: Apache-2.0
 

--- a/test/provisioner/provision_ibmcloud.go
+++ b/test/provisioner/provision_ibmcloud.go
@@ -1,5 +1,3 @@
-//go:build ibmcloud
-
 // (C) Copyright Confidential Containers Contributors
 // SPDX-License-Identifier: Apache-2.0
 

--- a/test/provisioner/provision_libvirt.go
+++ b/test/provisioner/provision_libvirt.go
@@ -1,4 +1,4 @@
-//go:build libvirt
+//go:build cgo
 
 // (C) Copyright Confidential Containers Contributors
 // SPDX-License-Identifier: Apache-2.0

--- a/test/provisioner/provision_libvirt_nocgo.go
+++ b/test/provisioner/provision_libvirt_nocgo.go
@@ -1,0 +1,10 @@
+//go:build !cgo
+
+// (C) Copyright Confidential Containers Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package provisioner
+
+func NewLibvirtProvisioner(properties map[string]string) (CloudProvisioner, error) {
+	panic("CGO is not enabled")
+}

